e2e/vmss.go

```diff
@@ -87,16 +55,59 @@ func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName str
 		model,
 		nil,
 	)
-	if err != nil {
-		return nil, fmt.Errorf("begin create VMSS %q: %w", vmssName, err)
+	var respErr *azcore.ResponseError
+	// sometimes the SKU is not available and we can't do anything. Skip the test in this case.
+	if config.SkipTestsWithSKUCapacityIssue && errors.As(err, &respErr) && respErr.StatusCode == 409 && respErr.ErrorCode == "SkuNotAvailable" {
```

Comment:
extract logic to check for `SkuNotAvailable` to a helper to make this a bit more readable?